### PR TITLE
Add `readiness_checks` to `google_workstations_workstation_config`

### PR DESCRIPTION
**Pull request by bschaatsbergen**
Adds the `readiness_checks` argument block to the `google_workstations_workstation_config` resource.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
workstations: added `readiness_checks` argument block to `google_workstations_workstation_config` resource (beta)
```

Relates https://github.com/hashicorp/terraform-provider-google/issues/16997

### CODE REVIEW

File: mmv1/third_party/terraform/services/workstations/resource_workstations_workstation_config_test.go.erb [hunk 1]
```diff
@@ -372,6 +372,83 @@ func testAccWorkstationsWorkstationConfig_disableTcpConnections(context map[stri
...
+  }
+  
+  resource "google_workstations_workstation_config" "default" {
+    provider               = google-beta
+    workstation_config_id  = "tf-test-workstation-config%{random_suffix}"
+    workstation_cluster_id = google_workstations_workstation_cluster.default.workstation_cluster_id
+    location               = "us-central1"
+
+    readiness_checks {
+      port = 22
```
Comment (bschaatsbergen) [inline mmv1/third_party/terraform/services/workstations/resource_workstations_workstation_config_test.go.erb:439]: According to https://cloud.google.com/workstations/docs/ssh-support
>All preconfigured Cloud Workstations images include an SSH server that runs on workstation port 22

Comment (bschaatsbergen) [reply mmv1/third_party/terraform/services/workstations/resource_workstations_workstation_config_test.go.erb:439]: So I'm hoping this works :)

**Comment by shuyama1** [comment]
Test failed with error:
```
Error: Error creating WorkstationConfig: googleapi: Error 400: readiness_check.path cannot be empty
```

**Comment by bschaatsbergen** [comment]
> Test failed with error:
> 
> ```
> 
> Error: Error creating WorkstationConfig: googleapi: Error 400: readiness_check.path cannot be empty
> 
> ```

Thanks for the quick feedback, I'll try '/' and port 80 ad that's apparently the default path and port.

**Comment by bschaatsbergen** [comment]
@shuyama1 addressed in 423987b

File: mmv1/products/workstations/WorkstationConfig.yaml [hunk 2]
```diff
@@ -428,6 +428,20 @@ properties:
...
+    name: 'readinessChecks'
+    description: |
+      Readiness checks to be performed on a workstation.
+    item_type: !ruby/object:Api::Type::NestedObject
+      properties:
+        - !ruby/object:Api::Type::String
+          name: 'path'
+          description: |
+            Path to which the request should be sent.
+        - !ruby/object:Api::Type::String
```
Comment (shuyama1) [inline mmv1/products/workstations/WorkstationConfig.yaml:441]:
```suggestion
        - !ruby/object:Api::Type::Integer
```
The provider crashes with error `readiness_checks.0.port: '' expected type 'string', got unconvertible type 'float64', value: '80'`

Comment (bschaatsbergen) [reply mmv1/products/workstations/WorkstationConfig.yaml]: Woops, thanks! Fixed.

File: mmv1/products/workstations/WorkstationConfig.yaml [hunk 3]
```diff
@@ -428,6 +428,20 @@ properties:
...
+    item_type: !ruby/object:Api::Type::NestedObject
+      properties:
+        - !ruby/object:Api::Type::String
+          name: 'path'
+          description: |
+            Path to which the request should be sent.
+        - !ruby/object:Api::Type::Integer
+          name: 'port'
+          description: |
+            Port to which the request should be sent.
```
Comment (shuyama1) [inline mmv1/products/workstations/WorkstationConfig.yaml:444]: I wonder if we want to mark these two fields required based on the previous error we ran into, even if they seem to be optional based on [the API doc](https://cloud.google.com/workstations/docs/reference/rest/v1beta/projects.locations.workstationClusters.workstationConfigs#readinesscheck). Changing a field from `required` to `optional` is non-changing while the other way around is breaking.

Plus, are these updatable? If so, can you add an update test following https://googlecloudplatform.github.io/magic-modules/develop/test/#add-an-update-test. Otherwise, we need to mark them `immutable`. Also, if you think the fields can be tested in any of the existing tests, feel free to add them to an existing one instead of creating a new test for this feature. Thanks

**Review by shuyama1 — COMMENTED** [review]
Mostly good. Some small comments

File: mmv1/products/workstations/WorkstationConfig.yaml (hunk 3 above)
Comment (bschaatsbergen) [reply mmv1/products/workstations/WorkstationConfig.yaml:444]: Great suggestion Shuya! I'll make them required, based on the previous error we faced. I would assume they are updatable, I'll introduce a test.

Comment (bschaatsbergen) [reply mmv1/products/workstations/WorkstationConfig.yaml]: Added `readiness_checks` to the basic example, do you think this would be enough? The basic example is part of the existing update test.

Comment (shuyama1) [reply mmv1/products/workstations/WorkstationConfig.yaml]: Would it be possible if we choose another update test? We try to keep only minimum required fields in the basic example config. Sorry about it

Comment (bschaatsbergen) [reply mmv1/products/workstations/WorkstationConfig.yaml]: I don't have an easy way to expose a different port, it's at least not possible via the existing arguments. The default exposed port is port 8080. Could we perhaps omit the update test?

**Comment by bschaatsbergen** [comment]
@shuyama1 any chance we can ship this :)?

**Review by shuyama1 — APPROVED** [review]
Thank you!
